KabuKit/Classes/core/Swift/SceneSequence.swift

```diff
@@ -9,85 +9,103 @@ internal var procedureByScene = [ScreenHashWrapper : TransitionProcedure]()
  このクラスの `start:` を明示的に呼ばないと内部でのSceneがリクエストしても画面遷移は行われない
 
  */
-public class SceneSequence<ContextType, GuideType: Guide> : Scene, SceneContainer {
+public class SceneSequence<FirstScene: Scene, Guide: SequenceGuide> : SceneCollection<Guide.Stage>, Scene, ScreenContainer {
 
-    public typealias Context = ContextType
-    
-    public typealias StageType = GuideType.Stage
-    
-    private var scenes: [Screen]
+
+    public typealias Context = FirstScene.Context
+    public typealias ReturnValue = FirstScene.ReturnValue
+    public typealias StageType = Guide.Stage
     
-    private var currentTransitionProcedure: TransitionProcedure?
+    typealias Stage = Guide.Stage
     
     private let isRecordable: Bool
-    
-    private var stage: StageType?
-    
-    private var guide: GuideType?
-    
-    private let dispatchQueue: DispatchQueue
+    private let firstScene: FirstScene
+    private let guide: Guide
+    private let leaveFunc: ((Guide.Stage, [Screen], ReturnValue?) -> Void)?
+    private let invokeFunc: (Guide.Stage, FirstScene) -> (() -> Void)?
```

Comment:
I felt that this variable name is too ambiguous.
(And I couldn't understand the task for this variable clearly... Could you give me some explanation?)

Reply:
`invokeFunc` is called at once when `SceneSequence`'s activate method is called and isStarted is false.
In a one word, this callback function supply `SceneSequence`'s initialize process.

so, maybe `initialize` fit better than invoke.

Reply:
and now,
maybe, leaveFunc is not necessary